errors/stackframe.go

```diff
@@ -31,7 +30,18 @@ func NewStackFrame(pc uintptr) (frame StackFrame) {
 	// and we want to show the line that corresponds to the function call
 	frame.File, frame.LineNumber = frame.Func().FileLine(pc - 1)
 	return
+}
 
+// NewStackFrameFromRuntime populates a stack frame object from a runtime.Frame object.
+func NewStackFrameFromRuntime(frame runtime.Frame) StackFrame {
+	pkg, name := packageAndName(frame.Func)
```

Comment:
It seems that frame.Func can still be nil sometimes. Might be worth attempting to salvage something from frame.Function in that case.

Reply:
Yes, that seems reasonable. I've pushed a commit that does this.